srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java

```
@@ -146,6 +169,51 @@ public TimeSeriesAnalysisSummary getAnalysisSummary(List<String> verificationJob
         .build();
   }
 
+  @Override
+  public MetricsAnalysisOverview getMetricsAnalysisOverview(String verifyStepExecutionId) {
+    VerificationJobInstance verificationJobInstance =
+        verificationJobInstanceService.getVerificationJobInstance(verifyStepExecutionId);
+    List<TransactionMetricInfo> transactionMetricInfos = getMetrics(verificationJobInstance.getAccountId(),
+        verifyStepExecutionId, DeploymentTimeSeriesAnalysisFilter.builder().build());
+    int noAnalysis = 0;
+    int healthy = 0;
+    int warning = 0;
+    int unhealthy = 0;
+    for (TransactionMetricInfo transactionMetricInfo : transactionMetricInfos) {
+      if (Objects.isNull(transactionMetricInfo.getTransactionMetric().getRisk())) {
+        noAnalysis++;
+        continue;
+      }
+      switch (transactionMetricInfo.getTransactionMetric().getRisk()) {
+        case NO_DATA:
+          noAnalysis++;
+          break;
+        case NO_ANALYSIS:
+          noAnalysis++;
+          break;
+        case HEALTHY:
+          healthy++;
+          break;
+        case OBSERVE:
+          warning++;
+          break;
+        case NEED_ATTENTION:
+          warning++;
+          break;
+        case UNHEALTHY:
+          unhealthy++;
```

we should throw an exception with default if it's guaranteed to match something.